### PR TITLE
Use Secret instead of ConfigMap for Ignition Status

### PR DESCRIPTION
**Pull request by stone-z**
Toward https://github.com/giantswarm/giantswarm/issues/8643 and https://github.com/giantswarm/giantswarm/issues/8644

@MarcelMue ConfigMap has a Data and BinaryData, Secret has a Data and StringData -- better to store it as string or bytes?

### CODE REVIEW

File: service/controller/resource/templateignition/create.go [hunk 1]
```diff
@@ -35,35 +35,34 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
+	s := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: key.StatusConfigMapName(cr.Spec.ClusterID),
 			Labels: map[string]string{
 				"cluster":       cr.Spec.ClusterID,
 				label.ManagedBy: project.Name(),
 			},
 		},
-		Data: map[string]string{
+		StringData: map[string]string{ // Data?
```
Comment (MarcelMue) [inline service/controller/resource/templateignition/create.go]: Let's change this to match upstream:
https://master.cluster-api.sigs.k8s.io/developer/providers/bootstrap.html#bootstrap-secret

So only a single value called `value` of type string.

File: service/controller/resource/templateignition/create.go [hunk 2]
```diff
@@ -35,35 +35,34 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	cm := &corev1.ConfigMap{
+	s := &corev1.Secret{
```
Comment (MarcelMue) [inline service/controller/resource/templateignition/create.go:40]: Let's also see how much you can align this with the secret in the spec in this section:
https://github.com/giantswarm/giantswarm/blob/5a871c1971b75043d8545b6f3fcc7456dc14ba67/specs/ignition_operator.md#ignition-cr-design

**Review by MarcelMue — CHANGES_REQUESTED** [review]

File: service/controller/resource/templateignition/create.go (hunk 1 above)
Comment (stone-z) [reply service/controller/resource/templateignition/create.go]: Done

File: service/controller/resource/templateignition/create.go (hunk 2 above)
Comment (stone-z) [reply service/controller/resource/templateignition/create.go:40]: Check it out, should match the spec now

File: service/controller/key/testdata/case-2-master-render.golden [hunk 3]
```diff
@@ -519,107 +519,4 @@ storage:
       contents:
         source: "data:text/plain;charset=utf-8;base64,YXBpVmVyc2lvbjogdjEKa2luZDogQ29uZmlnCnVzZXJzOgotIG5hbWU6IHNjaGVkdWxlcgogIHVzZXI6CiAgICBjbGllbnQtY2VydGlmaWNhdGU6IC9ldGMva3ViZXJuZXRlcy9zc2wvYXBpc2VydmVyLWNydC5wZW0KICAgIGNsaWVudC1rZXk6IC9ldGMva3ViZXJuZXRlcy9zc2wvYXBpc2VydmVyLWtleS5wZW0KY2x1c3RlcnM6Ci0gbmFtZTogbG9jYWwKICBjbHVzdGVyOgogICAgY2VydGlmaWNhdGUtYXV0aG9yaXR5OiAvZXRjL2t1YmVybmV0ZXMvc3NsL2FwaXNlcnZlci1jYS5wZW0KICAgIHNlcnZlcjogaHR0cHM6Ly9BUElEb21haW4KY29udGV4dHM6Ci0gY29udGV4dDoKICAgIGNsdXN0ZXI6IGxvY2FsCiAgICB1c2VyOiBzY2hlZHVsZXIKICBuYW1lOiBzZXJ2aWNlLWFjY291bnQtY29udGV4dApjdXJyZW50LWNvbnRleHQ6IHNlcnZpY2UtYWNjb3VudC1jb250ZXh0Cg=="
 
-    - path: /etc/kubernetes/encryption/k8s-encryption-config.yaml
```
Comment (MarcelMue) [inline service/controller/key/testdata/case-2-master-render.golden:522]: This seems wrong - the test should be altered to still render the master. We can take a look on how to implement this. Just noting that this is now called master render but renders a worker :)

File: service/controller/resource/templateignition/create.go [hunk 4]
```diff
@@ -25,45 +25,48 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
-		Data: map[string]string{
-			"master": master["."],
-			"worker": worker["."],
+		StringData: map[string]string{
+			"value": template["."],
 		},
 	}
 
-	actualCM, err := r.k8sClient.K8sClient().CoreV1().ConfigMaps(key.DefaultNamespace).Update(cm)
+	actualSecret, err := r.k8sClient.K8sClient().CoreV1().Secrets(key.DefaultNamespace).Update(s)
```
Comment (MarcelMue) [inline service/controller/resource/templateignition/create.go:56]: We will probably change this to place the secret in the same namespace as the CR - can do that separately though.

**Review by MarcelMue — APPROVED** [review]
Let's iterate.

**Pull request merged**